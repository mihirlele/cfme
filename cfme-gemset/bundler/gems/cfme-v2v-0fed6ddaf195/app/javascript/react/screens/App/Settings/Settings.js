import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb, Tabs, Tab } from 'patternfly-react';
import Toolbar from '../../../config/Toolbar';
import GeneralSettings from './screens/GeneralSettings';
import ConversionHostsSettings from './screens/ConversionHostsSettings';

const Settings = props => {
  const { match, redirectTo } = props;

  return (
    <React.Fragment>
      <Toolbar>
        <Breadcrumb.Item active>{__('Compute')}</Breadcrumb.Item>
        <Breadcrumb.Item active>{__('Migration')}</Breadcrumb.Item>
        <Breadcrumb.Item active>
          <strong>{__('Migration Settings')}</strong>
        </Breadcrumb.Item>
      </Toolbar>
      <div style={{ marginTop: 10 }}>
        <Tabs id="settings-tabs" activeKey={match.path} onSelect={key => redirectTo(key)} unmountOnExit>
          <Tab eventKey="/settings" title={__('Migration Throttling')}>
            <GeneralSettings />
          </Tab>
          <Tab eventKey="/settings/conversion_hosts" title={__('Conversion Hosts')}>
            <ConversionHostsSettings />
          </Tab>
        </Tabs>
      </div>
    </React.Fragment>
  );
};

Settings.propTypes = {
  match: PropTypes.object,
  redirectTo: PropTypes.func
};

export default Settings;
