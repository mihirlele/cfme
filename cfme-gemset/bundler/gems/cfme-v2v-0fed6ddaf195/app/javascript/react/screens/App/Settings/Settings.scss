.migration-settings h2 {
  margin-bottom: 25px;
}

.conversion-hosts-list .list-view-pf-main-info {
  padding: 10px 0;
}

.conversion-hosts-list .spinner.spinner-inline {
  margin-right: 10px;
}

.conversion-hosts-list-actions {
  min-width: 150px;
  text-align: right;
}

.postfix-label {
  background-color: #fafafa;
  border-left-width: 0px;
}
