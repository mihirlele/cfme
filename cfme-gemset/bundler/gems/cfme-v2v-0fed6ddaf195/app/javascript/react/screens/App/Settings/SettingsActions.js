import { reset } from 'redux-form';
import URI from 'urijs';
import API from '../../../../common/API';

import {
  V2V_FETCH_SERVERS,
  V2V_FETCH_SETTINGS,
  V2V_PATCH_SETTINGS,
  FETCH_V2V_CONVERSION_HOSTS,
  FETCH_V2V_CONVERSION_HOST_TASKS,
  SHOW_V2V_CONVERSION_HOST_WIZARD,
  HIDE_V2V_CONVERSION_HOST_WIZARD,
  V2V_CONVERSION_HOST_WIZARD_EXITED,
  POST_V2V_CONVERSION_HOSTS,
  SET_V2V_CONVERSION_HOST_TO_DELETE,
  SHOW_V2V_CONVERSION_HOST_DELETE_MODAL,
  HIDE_V2V_CONVERSION_HOST_DELETE_MODAL,
  DELETE_V2V_CONVERSION_HOST,
  SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  SHOW_V2V_CONVERSION_HOST_RETRY_MODAL,
  HIDE_V2V_CONVERSION_HOST_RETRY_MODAL,
  V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
} from './SettingsConstants';
import { getApiSettingsFromFormValues } from './helpers';
import { stepIDs } from './screens/ConversionHostsSettings/components/ConversionHostWizard/ConversionHostWizardConstants';

const _getServersActionCreator = url => dispatch =>
  dispatch({
    type: V2V_FETCH_SERVERS,
    payload: API.get(url)
  });

export const fetchServersAction = url => {
  const uri = new URI(url);
  return _getServersActionCreator(uri.toString());
};

const _getSettingsActionCreator = url => dispatch =>
  dispatch({
    type: V2V_FETCH_SETTINGS,
    payload: API.get(url)
  });

export const fetchSettingsAction = url => {
  const uri = new URI(url);
  return _getSettingsActionCreator(uri.toString());
};

const _patchSettingsActionCreator = (urls, settingsFormValues) => dispatch => {
  const patchBody = getApiSettingsFromFormValues(settingsFormValues);
  return dispatch({
    type: V2V_PATCH_SETTINGS,
    payload: new Promise((resolve, reject) => {
      Promise.all(urls.map(url => API.patch(url, patchBody)))
        .then(payloads => resolve(payloads[0]))
        .catch(e => reject(e));
    })
  });
};

export const patchSettingsAction = (servers, newSettings) => {
  const settingsUrls = servers.map(server => new URI(`${server.href}/settings`).toString());
  return _patchSettingsActionCreator(settingsUrls, newSettings);
};

const _getConversionHostsActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_CONVERSION_HOSTS,
    payload: API.get(url)
  });

export const fetchConversionHostsAction = url => {
  const uri = new URI(url);
  return _getConversionHostsActionCreator(uri.toString());
};

const _getConversionHostTasksActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_CONVERSION_HOST_TASKS,
    payload: API.get(url)
  });

export const fetchConversionHostTasksAction = url => {
  const uri = new URI(url);
  return _getConversionHostTasksActionCreator(uri.toString());
};

export const showConversionHostWizardAction = () => dispatch => dispatch({ type: SHOW_V2V_CONVERSION_HOST_WIZARD });

export const hideConversionHostWizardAction = () => dispatch => dispatch({ type: HIDE_V2V_CONVERSION_HOST_WIZARD });

export const conversionHostWizardExitedAction = () => dispatch => {
  dispatch({ type: V2V_CONVERSION_HOST_WIZARD_EXITED });
  // Dispatch reset for all the wizard step forms here
  Object.values(stepIDs).forEach(formName => {
    dispatch(reset(formName));
  });
};

const _postConversionHostsActionCreator = (url, postBodies) => dispatch =>
  dispatch({
    type: POST_V2V_CONVERSION_HOSTS,
    payload: Promise.all(postBodies.map(body => API.post(url, body)))
  });

export const postConversionHostsAction = (url, postBodies) =>
  _postConversionHostsActionCreator(new URI(url).toString(), postBodies);

export const setHostToDeleteAction = host => ({
  type: SET_V2V_CONVERSION_HOST_TO_DELETE,
  payload: host
});

export const showConversionHostDeleteModalAction = () => ({
  type: SHOW_V2V_CONVERSION_HOST_DELETE_MODAL
});

export const hideConversionHostDeleteModalAction = () => ({
  type: HIDE_V2V_CONVERSION_HOST_DELETE_MODAL
});

export const _deleteConversionHostActionCreator = (url, host) => dispatch =>
  dispatch({
    type: DELETE_V2V_CONVERSION_HOST,
    payload: new Promise((resolve, reject) => {
      API.post(`${url}/${host.id}`, {
        action: 'delete'
      })
        .then(response => {
          resolve(response);
        })
        .catch(e => reject(e));
    })
  });

export const deleteConversionHostAction = (url, host) =>
  _deleteConversionHostActionCreator(new URI(url).toString(), host);

export const setConversionHostTaskToRetryAction = task => ({
  type: SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  payload: task
});

export const showConversionHostRetryModalAction = () => ({
  type: SHOW_V2V_CONVERSION_HOST_RETRY_MODAL
});

export const hideConversionHostRetryModalAction = () => ({
  type: HIDE_V2V_CONVERSION_HOST_RETRY_MODAL
});

export const conversionHostRetryModalExitedAction = () => ({
  type: V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
});
