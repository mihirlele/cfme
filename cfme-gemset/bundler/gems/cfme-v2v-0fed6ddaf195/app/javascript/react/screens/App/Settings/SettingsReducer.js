import Immutable from 'seamless-immutable';

import {
  V2V_FETCH_SERVERS,
  V2V_FETCH_SETTINGS,
  V2V_PATCH_SETTINGS,
  FETCH_V2V_CONVERSION_HOSTS,
  FETCH_V2V_CONVERSION_HOST_TASKS,
  SHOW_V2V_CONVERSION_HOST_WIZARD,
  HIDE_V2V_CONVERSION_HOST_WIZARD,
  V2V_CONVERSION_HOST_WIZARD_EXITED,
  POST_V2V_CONVERSION_HOSTS,
  SET_V2V_CONVERSION_HOST_TO_DELETE,
  SHOW_V2V_CONVERSION_HOST_DELETE_MODAL,
  HIDE_V2V_CONVERSION_HOST_DELETE_MODAL,
  DELETE_V2V_CONVERSION_HOST,
  SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  SHOW_V2V_CONVERSION_HOST_RETRY_MODAL,
  HIDE_V2V_CONVERSION_HOST_RETRY_MODAL,
  V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
} from './SettingsConstants';

import {
  getFormValuesFromApiSettings,
  parseConversionHostTasksMetadata,
  indexConversionHostTasksByResource
} from './helpers';

export const initialState = Immutable({
  conversionHosts: [],
  conversionHostTasks: [],
  conversionHostTasksByResource: {},
  conversionHostToDelete: null,
  conversionHostDeleteModalVisible: false,
  conversionHostRetryModalMounted: false,
  conversionHostRetryModalVisible: false,
  conversionHostTaskToRetry: null,
  conversionHostWizardMounted: false,
  conversionHostWizardVisible: false,
  errorDeleteConversionHost: false,
  errorFetchingConversionHosts: null,
  errorFetchingConversionHostTasks: null,
  errorFetchingServers: null,
  errorFetchingSettings: null,
  errorPostingConversionHosts: null,
  errorSavingSettings: null,
  fetchingServersRejected: false,
  fetchingSettingsRejected: false,
  isDeletingConversionHost: false,
  isFetchingConversionHosts: false,
  isFetchingConversionHostTasks: false,
  isFetchingServers: false,
  isFetchingSettings: false,
  isPostingConversionHosts: false,
  isRejectedDeletingConversionHost: false,
  isRejectedFetchingConversionHosts: false,
  isRejectedFetchingConversionHostTasks: false,
  isRejectedPostingConversionHosts: false,
  isSavingSettings: false,
  postConversionHostsResults: [],
  savedSettings: {},
  savingSettingsRejected: false,
  servers: []
});

export default (state = initialState, action) => {
  switch (action.type) {
    case `${V2V_FETCH_SERVERS}_PENDING`:
      return state
        .set('isFetchingServers', true)
        .set('fetchingServersRejected', false)
        .set('errorFetchingServers', null);
    case `${V2V_FETCH_SERVERS}_REJECTED`:
      return state
        .set('isFetchingServers', false)
        .set('fetchingServersRejected', true)
        .set('errorFetchingServers', action.payload);
    case `${V2V_FETCH_SERVERS}_FULFILLED`:
      return state
        .set('isFetchingServers', false)
        .set('fetchingServersRejected', false)
        .set('errorFetchingServers', null)
        .set('servers', action.payload.data.resources);

    case `${V2V_FETCH_SETTINGS}_PENDING`:
      return state
        .set('isFetchingSettings', true)
        .set('fetchingSettingsRejected', false)
        .set('errorFetchingSettings', null);
    case `${V2V_FETCH_SETTINGS}_REJECTED`:
      return state
        .set('isFetchingSettings', false)
        .set('fetchingSettingsRejected', true)
        .set('errorFetchingSettings', action.payload);
    case `${V2V_FETCH_SETTINGS}_FULFILLED`:
      return state
        .set('isFetchingSettings', false)
        .set('fetchingSettingsRejected', false)
        .set('errorFetchingSettings', null)
        .set('savedSettings', getFormValuesFromApiSettings(action.payload.data));

    case `${V2V_PATCH_SETTINGS}_PENDING`:
      return state
        .set('isSavingSettings', true)
        .set('savingSettingsRejected', false)
        .set('errorSavingSettings', null);
    case `${V2V_PATCH_SETTINGS}_REJECTED`:
      return state
        .set('isSavingSettings', false)
        .set('savingSettingsRejected', true)
        .set('errorSavingSettings', action.payload);
    case `${V2V_PATCH_SETTINGS}_FULFILLED`:
      return state
        .set('isSavingSettings', false)
        .set('savingSettingsRejected', false)
        .set('errorSavingSettings', null)
        .set('savedSettings', getFormValuesFromApiSettings(action.payload));

    case `${FETCH_V2V_CONVERSION_HOSTS}_PENDING`:
      return state
        .set('isFetchingConversionHosts', true)
        .set('isRejectedFetchingConversionHosts', false)
        .set('errorFetchingConversionHosts', null);
    case `${FETCH_V2V_CONVERSION_HOSTS}_FULFILLED`:
      return state
        .set('conversionHosts', action.payload.data.resources)
        .set('isFetchingConversionHosts', false)
        .set('isRejectedFetchingConversionHosts', false)
        .set('errorFetchingConversionHosts', null);
    case `${FETCH_V2V_CONVERSION_HOSTS}_REJECTED`:
      return state
        .set('isFetchingConversionHosts', false)
        .set('isRejectedFetchingConversionHosts', true)
        .set('errorFetchingConversionHosts', action.payload);

    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_PENDING`:
      return state
        .set('isFetchingConversionHostTasks', true)
        .set('isRejectedFetchingConversionHostTasks', false)
        .set('errorFetchingConversionHostTasks', null);
    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_FULFILLED`: {
      const tasksWithMetadata = parseConversionHostTasksMetadata(action.payload.data.resources);
      const tasksByResource = indexConversionHostTasksByResource(tasksWithMetadata);
      return state
        .set('conversionHostTasks', tasksWithMetadata)
        .set('conversionHostTasksByResource', tasksByResource)
        .set('isFetchingConversionHostTasks', false)
        .set('isRejectedFetchingConversionHostTasks', false)
        .set('errorFetchingConversionHostTasks', null);
    }
    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_REJECTED`:
      return state
        .set('isFetchingConversionHostTasks', false)
        .set('isRejectedFetchingConversionHostTasks', true)
        .set('errorFetchingConversionHostTasks', action.payload);

    case SHOW_V2V_CONVERSION_HOST_WIZARD:
      return state.set('conversionHostWizardMounted', true).set('conversionHostWizardVisible', true);
    case HIDE_V2V_CONVERSION_HOST_WIZARD:
      return state.set('conversionHostWizardVisible', false);
    case V2V_CONVERSION_HOST_WIZARD_EXITED:
      return state.set('conversionHostWizardMounted', false);

    case `${POST_V2V_CONVERSION_HOSTS}_PENDING`:
      return state
        .set('isPostingConversionHosts', true)
        .set('isRejectedPostingConversionHosts', false)
        .set('errorPostingConversionHosts', false);
    case `${POST_V2V_CONVERSION_HOSTS}_FULFILLED`:
      return state
        .set('isPostingConversionHosts', false)
        .set('isRejectedPostingConversionHosts', false)
        .set('errorPostingConversionHosts', null)
        .set('postConversionHostsResults', action.payload);
    case `${POST_V2V_CONVERSION_HOSTS}_REJECTED`:
      return state
        .set('isPostingConversionHosts', false)
        .set('isRejectedPostingConversionHosts', true)
        .set('errorPostingConversionHosts', action.payload);
    case SET_V2V_CONVERSION_HOST_TO_DELETE:
      return state.set('conversionHostToDelete', action.payload);
    case SHOW_V2V_CONVERSION_HOST_DELETE_MODAL:
      return state.set('conversionHostDeleteModalVisible', true);
    case HIDE_V2V_CONVERSION_HOST_DELETE_MODAL:
      return state.set('conversionHostDeleteModalVisible', false);

    case `${DELETE_V2V_CONVERSION_HOST}_PENDING`:
      return state.set('isDeletingConversionHost', action.payload);
    case `${DELETE_V2V_CONVERSION_HOST}_FULFILLED`:
      return state
        .set('deleteConversionHostResponse', action.payload.data)
        .set('isDeletingConversionHost', null)
        .set('isRejectedDeletingConversionHost', false)
        .set('errorDeleteConversionHost', null)
        .set('conversionHostDeleteModalVisible', false);
    case `${DELETE_V2V_CONVERSION_HOST}_REJECTED`:
      return state
        .set('errorDeleteConversionHost', action.payload)
        .set('isRejectedDeletingConversionHost', true)
        .set('isDeletingConversionHost', null)
        .set('conversionHostDeleteModalVisible', false);

    case SET_V2V_CONVERSION_HOST_TASK_TO_RETRY:
      return state.set('conversionHostTaskToRetry', action.payload);
    case SHOW_V2V_CONVERSION_HOST_RETRY_MODAL:
      return state.set('conversionHostRetryModalMounted', true).set('conversionHostRetryModalVisible', true);
    case HIDE_V2V_CONVERSION_HOST_RETRY_MODAL:
      return state.set('conversionHostRetryModalVisible', false);
    case V2V_CONVERSION_HOST_RETRY_MODAL_EXITED:
      return state.set('conversionHostRetryModalMounted', false);

    default:
      return state;
  }
};
