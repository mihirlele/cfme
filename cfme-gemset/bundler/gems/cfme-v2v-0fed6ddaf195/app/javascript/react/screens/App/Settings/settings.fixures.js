import Immutable from 'seamless-immutable';

export const servers = Immutable({
  name: 'servers',
  count: 2,
  subcount: 2,
  pages: 1,
  resources: [
    {
      href: 'http://0.0.0.0:8080/api/servers/42000000000001'
    },
    {
      href: 'http://0.0.0.0:8080/api/servers/42000000000002'
    }
  ]
});

export const settings = Immutable({
  transformation: {
    limits: {
      max_concurrent_tasks_per_host: 10,
      max_concurrent_tasks_per_ems: 10
    }
  },
  otherSettings: {
    mock: 'data'
  }
});

export const settingsFormValues = Immutable({
  max_concurrent_tasks_per_host: 10,
  max_concurrent_tasks_per_ems: 10
});

export const fetchServersData = {
  method: 'GET',
  fetchServersUrl: '/api/servers?expand=resources',
  response: { data: servers }
};

export const fetchSettingsData = {
  method: 'GET',
  fetchSettingsUrl: '/api/settings',
  response: { data: settings }
};

export const patchSettingsData = {
  method: 'PATCH',
  response: { data: settings }
};
