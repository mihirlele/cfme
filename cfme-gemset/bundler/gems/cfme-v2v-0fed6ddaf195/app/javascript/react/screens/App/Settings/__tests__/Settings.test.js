import React from 'react';
import { shallow } from 'enzyme';
import Settings from '../Settings';

describe('Settings component', () => {
  it('renders correctly', () => {
    const component = shallow(<Settings match={{ path: '/settings' }} />);
    expect(component).toMatchSnapshot();
  });
});
