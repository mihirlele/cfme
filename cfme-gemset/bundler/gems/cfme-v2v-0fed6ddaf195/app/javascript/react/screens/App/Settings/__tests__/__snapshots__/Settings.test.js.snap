// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<React.Fragment>
  <Toolbar>
    <BreadcrumbItem
      active={true}
    >
      Compute
    </BreadcrumbItem>
    <BreadcrumbItem
      active={true}
    >
      Migration
    </BreadcrumbItem>
    <BreadcrumbItem
      active={true}
    >
      <strong>
        Migration Settings
      </strong>
    </BreadcrumbItem>
  </Toolbar>
  <div
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Uncontrolled(Tabs)
      activeKey="/settings"
      id="settings-tabs"
      onSelect={[Function]}
      unmountOnExit={true}
    >
      <Tab
        eventKey="/settings"
        title="Migration Throttling"
      >
        <Connect(ReduxForm) />
      </Tab>
      <Tab
        eventKey="/settings/conversion_hosts"
        title="Conversion Hosts"
      >
        <Connect(ConversionHostsSettings) />
      </Tab>
    </Uncontrolled(Tabs)>
  </div>
</React.Fragment>
`;
